### PR TITLE
Playback rate

### PR DESCRIPTION
**Pull request by pcbeard**
This builds on #262, adding overlap property for controlling the quality.

### CODE REVIEW

**Comment by adamontherun** [comment]
tested this tonight and seems to work really well. 
@tumtumtum if you have time, could you please review and consider this feature? Variable speed audio is the number one request feature in my podcast player that I don't have implemented yet.

thanks
adam

**Comment by dfg-nota** [comment]
This feature is a requirement for my current project and the only reason why I can't use StreamingKit right now. Would be nice to see this merged at some point.

**Comment by qianbodong** [comment]
How to Add   “pan”

**Comment by prakashdesai** [comment]
@qianbodong did you find the way to add the pan? i am also looking for same thing.

**Comment by cbttpaul** [comment]
Would love to see this get merged...

**Comment by gaodeng** [comment]
any news on this?

File: StreamingKit/StreamingKit/STKAudioPlayer.m [hunk 1]
```diff
@@ -2133,6 +2169,22 @@ -(void) createAudioConverter:(AudioStreamBasicDescription*)asbd
...
+    CHECK_STATUS_AND_RETURN(AUGraphNodeInfo(audioGraph, playbackRateNode, &playbackRateUnitDescription, &playbackRateUnit));
+    
+    //maxframes changed here
+    CHECK_STATUS_AND_RETURN(AudioUnitSetProperty(playbackRateUnit, kAudioUnitProperty_MaximumFramesPerSlice, kAudioUnitScope_Global, 0, &maxFramesPerSlice, sizeof(maxFramesPerSlice)));
+#if TARGET_OS_IPHONE
+    CHECK_STATUS_AND_RETURN(AudioUnitSetParameter(playbackRateUnit, kNewTimePitchParam_Rate, kAudioUnitScope_Global, 0, 1, 0));
+#endif
+    
+  CHECK_STATUS_AND_RETURN(AudioUnitSetProperty(playbackRateUnit, kAudioUnitProperty_StreamFormat, kAudioUnitScope_Input, kOutputBus, &canonicalAudioStreamBasicDescription, sizeof(canonicalAudioStreamBasicDescription)));
+}
```
Comment (revolter) [inline StreamingKit/StreamingKit/STKAudioPlayer.m]: This exact method introduces a crash in some rare cases (but reproducible):

image

Do you have any idea of why would it crash there?

Comment (revolter) [reply StreamingKit/StreamingKit/STKAudioPlayer.m]: @pcbeard, Do you have any idea?

**Pull request merged**